tools/code-owners-parser/Azure.Sdk.Tools.RetrieveCodeOwners.Tests/Azure.Sdk.Tools.RetrieveCodeOwners.Tests.csproj

```diff
@@ -2,6 +2,7 @@
 
   <PropertyGroup>
     <TargetFramework>net6.0</TargetFramework>
+    <LangVersion>10.0</LangVersion>
```

Comment:
Is this necessary to specify? What is the default language feature?

Reply:
@weshaggard Removing this makes the projects revert to C# 8, which is unclear to me why, because [they should default to C# 10](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version#defaults). Thus I force it to C# 10 to be able to use bunch of newer features from it. On C# 8 I end up with bunch of compilation errors (unsurprisingly):

![image](https://user-images.githubusercontent.com/4429827/212440219-92599457-4b61-4365-8d50-01e5ea62fc5e.png)

If you have an idea why the default is 8 and not 10, let me know.

Reply:
@weshaggard A ha! I think it is coming from [Directory.Build.props](https://github.com/Azure/azure-sdk-tools/blob/c124d9b0eec9927f28613747254d01f51dc2fb84/eng/Directory.Build.props#L7) ! I will make a PR to update this to `10.0`, now that we have moved everything to `net6.0`.

Reply:
Submitted PR:
- #5175